### DISCUSSION TITLE
Convert fundraising remainder address(es) to keyless account actor(s) (FIP-0096)

### DISCUSSION THREAD

**Discussion by Fatman13** [filecoin-project/FIPs#1033] [Enhancements - Technical]
## 中文 tldr

全网 2.5% 原用于融资的通证进行锁仓处理，日后具体用处由 FIP 决定。

## Change Proposed

Similar to #901, covert fundraising remainder address(es) type from a multisig to account(s) that is keyless (like f099).

## Motivation 

Like how mining reserve is made into a keyless account and ready to be [potentially burned](https://github.com/filecoin-project/FIPs/discussions/1030), fundraising remainder address(es) is also in the linger troubled by similar dilemma described in motivation section of #901. We compiled the following 4 motivations for the proposal...

### 1. follow the spec

Mining reserve in the Filecoin spec is described as the following...

> It will be up to the community to determine in the future how to distribute those tokens, through Filecoin improvement proposals (FIPs) or similar decentralized decision-making processes.
> -- excerpts from #901

[fundraising remainder](https://spec.filecoin.io/#section-systems.filecoin_token.token_allocation) in the Filecoin spec is described as the following...

> Of the Filecoin genesis block allocation, 10% of FIL_BASE were allocated for fundraising, of which 7.5% were sold in the 2017 token sale, and the 2.5% remaining were allocated for ecosystem development and potential future fundraising.

Though paragraph about fundraising remainder didn't specifically articulate "up to the community to determine" as in mining reserve, by virtue of us being a crypto / blockchain community it can be assumed **by default** it is up to the community, ie FIP process, when no governance body was mentioned to oversight the distribution.

### 2. governance

> It is worth noting that when the spec was written and when the network was first launched, the FIP process was not yet well-designed or matured to support network governance, and significant improvements have been made since then. 
> -- excerpts from #901

Similar to the point #901 made, this proposal is to amend the inadequacy that network in its early days has neglected. 

### 3. security

> This also creates a security black hole in the network where if 2 out of the 3 signer keys of f090 are compromised by malicious actors, they may cause serious economic damage to the Filecoin network. 
> -- excerpts from #901

fundraising remainder address(es) face identical security challenges as mining reserve too.

### 4. operation

> In addition, having the reserve holds in a multisig also means changes toward the mining reserve..., may need to be managed by the msig signers via signed transactions which is a huge operational overhead (and again against the principle).
> -- excerpts from #901

fundraising remainder address(es) face identical operation challenges as mining reserve too.

## Appendix

![image](https://github.com/user-attachments/assets/5728f395-c7ad-410f-87a9-3c106add1dfb)

*Ideas stemmed from discussion [here](https://github.com/filecoin-project/FIPs/discussions/887#discussioncomment-7859466), [here](https://github.com/filecoin-project/FIPs/discussions/901#discussioncomment-8586779), [here](https://filecoinproject.slack.com/archives/C01EU76LPCJ/p1708912934208939?thread_ts=1708560251.653839&cid=C01EU76LPCJ) and built upon #901*
*A proposal by Ancients Research team*

**Comment by jennijuju** [comment]
n00b q - what is the address of this fund?

**Reply by Fatman13** [reply]
We don't have the technical capacity to find out, but if anyone do find out we are happy to add them to authors list.

**Comment by ZenGround0** [comment]
I am in support of this proposal.

**Comment by jbenet** [comment]
As described in the original Filecoin SAFT sale documents of 2017, Protocol Labs, Inc. initially owned all of the tokens allotted to fundraising. PL sold SAFTs for most of this allocation to raise funding to develop the network in 2017. PL reserved the remainder for future fundraising, ecosystem development, and more. To be clear, PL has owned these tokens from the beginning, and has sole discretion over their use. That said, PL has chosen to allocate these tokens to various ecosystem growth endeavors over the years - including ecosystem development, ecosystem partner support, community grants, and Filecoin growth programs. You can read more [here](https://protocol.ai/blog/pl-participation-in-the-filecoin-economy/).

**Comment by ArthurWang1255** [comment]
I don't think there is a Bible in this community. The documents mentioned above were just white paper/spec written by PL few years earlier. Maybe the community agreed at the beginning, but it does not mean that the community still agrees. 

PL/the founder is just a member. Giving a large percentage of assets to a few people‘s discretion to dispose of has been proven to be inefficient, creating rent-seeking behaviors in the Filecoin community.

I don’t think PL has the divine right to control these tokens, but I am not saying PL must have no right to control these tokens either. However the right must be delegated to PL by the majority of community. That makes this discussion meaningful. 

Btw, thanks to this idea. It reminds me of the fact that in addition to the fund raising remainder mentioned here, PL's other vesting token is not determined by any god either. In view of the community's controversy over PL's incapability, a large number of SAFT vesting tokens that is lack of scrutinizing from the community can theoretically all be included in the scope of community discussion and adjustment.

**Reply by Fatman13** [reply]
Instead of being a Web3 / Crypto / blockchain community, it felt like the project is being run like a traditional top-down franchiser franchisee business (like KFC in NA) where the franchiser enjoys asset light, lower risks & higher margin while the franchisees bears asset heavy, higher risk & lower margin.

**Comment by alberto-gonzales** [comment]
HI.

@Fatman13 are there any plans to move this forward in the process or do you aim at leaving this in discussion stage?

**Reply by anorth** [reply]
I don't think there's much point. I'm pretty sure it's address https://filfox.info/en/address/f0122 and it only has 301 FIL left (See Juan's comment above).

**Reply by Fatman13** [reply]
i guess i could move discussion forward by submitting a PR. Though it maybe practically fruitless like what Alex has suggested above, but i think it could be symbolic. Worth a PR.

**Reply by Fatman13** [reply]
Just created a [PR](https://github.com/filecoin-project/FIPs/pull/1080). To see who are web3-minded rather than web2-minded and to see if spec is respected or not.

**Reply by alberto-gonzales** [reply]
will follow this closely @Fatman13

**Comment by anorth** [comment]
I've merged the draft of FIP-0096 https://github.com/filecoin-project/FIPs/pull/1080

**Comment by Tanisha-fil** [comment]
Hi all — How do we move this forward?

**Context:**

It was merged on Dec 12, 2024
Fully discussed and editorially approved

As discussed in the last Implementers' call, this FIP will need implementation efforts.

**Reply by rvagg** [reply]
> without requiring a network upgrade

This FIP requires a network upgrade. It needs a migration, like FIP-0085 ([like this](https://github.com/filecoin-project/go-state-types/blob/3cfb1b489520ab009985ac82557494cbe150becd/builtin/v14/migration/top.go#L99-L148)).

I think that means it's not valid for jumping straight to "Final".

It could just go through Last Call and core devs could consider the cost of doing a migration (it's not free - someone has to do the work, it has to be carefully reviewed and tested, and the network bears nontrivial risk for everything we add to migrations for upgrades) is worth locking up 301 FIL.

**Reply by ZenGround0** [reply]
> locking up 301 FIL

In light of the small number of tokens under discussion I think it is not practically useful to implement this FIP.

**Reply by Fatman13** [reply]
As mentioned in the [proposal](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0096.md), it doesn't represent utility but more on integrity and security aspect of the network. It is imperative to me and many crypto native that the code should follow the spec. Or we may open up the slippery slope of deciding things without network consensus which is not in the spirit of decentralization and Filecoin mission. To me not locking up the address should require an FIP to change the spec accordingly and not decided by discussion of core-devs.

**Reply by rvagg** [reply]
Two issues with this line of thinking that I can see that Core Devs et. al. would need to be considering (I'm not on Core Devs FWIW) with this proposal:

1. The fact that migrations like this aren't free, they require time to implement, time to review, time to test, time and care to observe on calibnet. Migrations involve risk, regardless of how trivial they are. This one has easy precedents so the actual implementation isn't hard at all, it's just not free and we should be careful considering something that looks trivial to be approximately-free.
2. As per [Juan's comment above](https://github.com/filecoin-project/FIPs/discussions/1033#discussioncomment-10099906), the claim is that this account was PL's in the first place, which is at odds with your claim that making this account keyless would "follow the spec", since presumably the remaining 301 FIL are still property of PL as per Juan's comment. I think that would be an important point for core devs or other reviewers to consider—making them unspendable by PL could be considered theft by many if they accept the premise—perhaps not a good precedent to set for a blockchain. PL could choose to grant this FIP passage for whatever reason, but that'd be up to them (I'm also not PL FWIW).

**Reply by Fatman13** [reply]
> 1. This one has easy precedents so the actual implementation isn't hard at all, it's just not free and we should be careful considering something that looks trivial to be approximately-free.

Yes, it requires core-dev's input to gauge the work load. No problem with that.

> 2. As per [Juan's comment](https://github.com/filecoin-project/FIPs/discussions/1033#discussioncomment-10099906), the claim is that this account was PL's in the first place, which is at odds with your claim that making this account keyless would "follow the spec", since presumably the remaining 301 FIL are still property of PL as per Juan's comment.

When in odds, let us consult the spec, the single source of truth. According to the spec, PL was given control of 15% of the allocation (not including the 2.5% fundraising remainder).

<img width="1610" height="1021" alt="image" src="https://github.com/user-attachments/assets/9883c8d8-bd45-4e2b-a88b-7867ea37911e" />
